src/main/java/seedu/cakecollate/ui/CommandBox.java

```diff
@@ -69,6 +75,68 @@ private void setStyleToIndicateCommandFailure() {
         styleClass.add(ERROR_STYLE_CLASS);
     }
 
+    /**
+     * Adds the command typed in by the user to the array list userInputs.
+     * @param input the command that the user has typed in.
+     */
+    public void updateUserInputs(String input) {
+        userInputs.add(input);
+        incrementUserInputsIndex();
+        userInputsIndex = userInputs.size() - 1;
+        firstDecrementAfterUserInput = true;
+        positionCaretInTheEnd();
+    }
+
+    private void incrementUserInputsIndex() {
+        if (userInputsIndex + 1 < userInputs.size() && userInputsIndex < userInputs.size()) {
+            userInputsIndex++;
+        }
+    }
+
+    private void decrementUserInputsIndex() {
+        if (!firstDecrementAfterUserInput && userInputsIndex > 0) {
+            userInputsIndex--;
+        }
+        firstDecrementAfterUserInput = false;
+    }
+
+    public void setCommandTextField(Optional<String> input) {
```

Comment:
Remember to add Javadoc comments for this method and also the other public methods,